### PR TITLE
[autoscaling] some doc updates

### PR DESCRIPTION
**Pull request by ericl**
cc @richardliaw @vlad17

### CODE REVIEW

**Review by richardliaw — COMMENTED** [review]
Looks fine to me; does vlad have comments?

File: doc/source/autoscaling.rst [hunk 1]
```diff
@@ -62,7 +67,7 @@ The setup commands you use should ideally be *idempotent*, that is, can be run m
 Syncing git branches
 --------------------
 
-A common use case is syncing a particular local git branch to all workers of the cluster. There is a nice way to do this as follows:
+A common use case is syncing a particular local git branch to all workers of the cluster. However, if you just put a `git checkout <branch>` in the setup commands, the autoscaler won't know when to rerun the command to pull in updates. There is a nice workaround for this by including the git SHA in the input (the hash of the file will change if the branch is updated):
```
Comment (vlad17) [inline doc/source/autoscaling.rst]: should maybe say something about head needing to be pushed on that branch

**Review by vlad17 — COMMENTED** [review]
setup_commands -- every cmd is on a fresh shell

File: doc/source/autoscaling.rst (hunk 1 above)
Comment (ericl) [reply doc/source/autoscaling.rst]: Done

File: doc/source/autoscaling.rst [hunk 2]
```diff
@@ -6,13 +6,13 @@ The ``ray create_or_update`` command starts an AWS Ray cluster from your persona
...
 
-First, ensure you have configured your AWS credentials in ``~/.aws/credentials``,
+First, install boto (``pip install boto3``) and configure your AWS credentials in ``~/.aws/credentials``,
 as described in `the boto docs <http://boto3.readthedocs.io/en/latest/guide/configuration.html>`__.
 
 Then you're ready to go. The provided `ray/python/ray/autoscaler/aws/example.yaml <https://github.com/ray-project/ray/tree/master/python/ray/autoscaler/aws/example.yaml>`__ cluster config file will create a small cluster with a m4.large head node (on-demand), and two m4.large `spot workers <https://aws.amazon.com/ec2/spot/>`__, configured to autoscale up to four m4.large workers.
 
 Try it out by running these commands from your personal computer. Once the cluster is started, you can then
-SSH into the head node to run Ray programs with ``ray.init(redis_address="<node_internal_ip>:6379")``.
+SSH into the head node to run Ray programs with ``ray.init(redis_address=ray.services.get_node_ip_address() + ":6379")``.
```
Comment (ericl) [inline doc/source/autoscaling.rst]: cc @pcmoritz

File: doc/source/autoscaling.rst [hunk 3]
```diff
@@ -27,6 +27,12 @@ SSH into the head node to run Ray programs with ``ray.init(redis_address="<node_
     # Teardown the cluster
     $ ray teardown ray/python/ray/autoscaler/aws/example.yaml
 
+To run connect to applications running on the cluster (e.g. Jupyter notebook) using a web browser, you can forward the port to your local machine using SSH:
+
+.. code-block:: bash
+
+    $ ssh -L 8899:localhost:8899 -i <key> <user>@<addr> jupyter notebook --port=8899
```
Comment (ericl) [inline doc/source/autoscaling.rst]: cc @pcmoritz

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
failures unrelated...

**Pull request merged**